frontend/packages/console-dynamic-plugin-sdk/src/webpack/ConsoleRemotePlugin.ts

```diff
@@ -52,6 +52,13 @@ export class ConsoleRemotePlugin {
       compiler.options.output.enabledLibraryTypes.push(remoteEntryLibraryType);
     }
 
+    compiler.options.plugins.push(
+      new webpack.IgnorePlugin({
```

Comment:
this is needed for the compilation of dynamic plugin.
Since we have a `require` calls in https://github.com/openshift/console/pull/9432/files#diff-84c872e4e52707828a4103b7a170f15538ead0269478510f7b944ca7d06cd822 we need to ignore them. They are properly resolved during the runtime

Reply:
Is the `IgnorePlugin` what ensures we don't package extra code in the plugin lib?

Reply:
no.
Sice we only transpile SDK via `tsc`, and do not use bundler, the `tsc` 'ignores' these `require` statements

The problem is when dynamic plugin is being bundled, `webpack` sees these `require` and complains that `@console/internal` is nowhere to be found so we add `IgnorePlugin` to tell `weback` to ignore the `require`

Reply:
Just to be clear then: a dynamic plugin won't ever package code from react, the plugin sdk, or any other lib we choose to override?

Reply:
> Just to be clear then: a dynamic plugin won't ever package code from react, the plugin sdk, or any other lib we choose to override?

Currently such modules are bundled (chunks created) as part of the plugin. At runtime, we instruct the entry chunk (`plugin-entry.js`) to use "our" (Console host app specific) runtime implementation of such modules.

Reply:
@vojtechszocs  How can we make it so that these libs don't make it at all in the plugin code?

Reply:
> @vojtechszocs How can we make it so that these libs don't make it at all in the plugin code?

We can do that by adjusting the way webpack builds dynamic plugins, i.e. via our webpack `ConsoleRemotePlugin`.

There should be no behavioral difference between "plugin-specific React module {present,absent} but overridden" cases.